### DISCUSSION TITLE
Definition of internal node quality score

### DISCUSSION THREAD

**Discussion by Robaina** [Robaina/MetaTag#39] [Ideas]
This thread is intended to propose ideas to design a "formula" to assign quality score values to internal nodes

**Comment by gecko1990** [comment]
I guess we would obtain the score from some mathematical estimation using the values of the leafs, right? Some points we might have to consider:

- Give more value to the leafs with experimental evidence

- Do we make a mean? Do we consider the branch length?

**Comment by JuanRivasSantisteban-zz** [comment]
These days I have been learning about genetic distances in trees produced by _Maximum Likehood_. I propose the following formula, as I found it very useful to propagate a function to a parent node. Of course, I could send some of the trees that I have calculated with this method, in case the explanation of the terms is not intuitive. I have made quite a few changes to the formula until it has come pretty close to a score that allows discriminating nodes with a high functional value from the functionally-ambiguous ones, although the "value" could only make biological sense with several important assumptions to consider: 

**Functional Node’s Score in a _Maximum Likehood_ tree:**

_Assuming that Log-Likehood value is the best possible for a tree;
Assuming that every sequence in the tree is biologically related (X squared composition test); 
Assuming all the scored leaves are true for the desired function:_

Ns = Cns - (1-f) · fd · (L / Lmax)


**Weighted Children Node’s score (Cns)**

Weighted mean between the first children Node’s Scores. Weight is stablished by the branch length fraction of total. First nodes in direct contact with leaf evidence scores are equal to 1.00. First nodes without contact with leaf evidence scores are equal to 0. 

Cns = ∑(i=1) v(i)·w(i) / ∑(i=1) w(i)

v: actual children node’s score, 
w: branch weight (b'/b), 
b': the opposite node’s branch length, 
b: the actual node’s branch length.

**Functional Evidence (f)**

Mean of every subordinated leaf score for functional evidence. 

**Functional divergence per node (fd)**

Every Node represent a divergence point between two primordial sequences, so differential value of branch lengths equals functional divergence value per node.

fd = LB/SB  ;

LB: longest branch to next node, SB: shortest branch to next node.

**Branch length (L)**

In a node, minimal genetic distance to next node with a functional score. This is a needed value to assess a divergence ratio between the next sequence divergence and the longest in the tree, returning a value ranging between 0 and 1.0. In other words, a normalized measure of proximity to the next node regarding desired function.

**Maximum branch length of the tree (Lmax)**

Longest genetic distance between two nodes in the whole tree.

PDF with the formulas:
[Functional_Node's_Score.pdf](https://github.com/Robaina/TRAITS/files/8023480/Functional_Node.s_Score.pdf)
